### PR TITLE
Phonebook

### LINKED ISSUE #448: Feature request: Self-serve DECT preregistration
***TL;DR***

POC team would like to offer pre-registration as a feature on the user profile at Bornhack web site.

# User story

As a hacker visiting camp, I want the Deluxe(tm) Seamless(r) experience of booting my DECT phone without all the hassle and away-from-keyboardness a visit to the POC may induce.

So, until the POC is fully integrated with the bar area I expect our robot overlords to handle all manual labor leading to my phone being ready for novely calls and ordering drinks.

# Background

To enable an automated flow for registration upon arrival, the POC needs:

- The extension number that was requested
- A verification token associated with the extension

The token will be used to register and activate the associated extension number with the device can provide it.

Like the Eventphone registration flow, we will use phone numbers as the verification token.

1. When an extension is reserved by a user, a longer (unique) phone number is generated and associated with the extension number requested.
2. At arrival to the camp, all DECT phones not yet verified are given throwaway numbers that are usable directly.
3. When the user dial the verification number associated with the extension actually requested, it will be activated and registered to the DECT phone that made the call.

The temporary numbers given at arrival have a separate dial plan that can only dial the automated activation numbers, no other handsets. Handsets are sandboxed this way until they are verified by the user (or blessed by the POC).

# Proposal

We would like to request two additional input fields for the user profile model:

- DECT extension (int)
- Publish in phonebook? (bool)

If possible, it would be really nice if there was a way to present the verification number on the user profile page as well.

Could this number be generated per-user in beforehand? It doesn't really have to be related to the actual extension number, since we do not plan to support multiple extensions per user at this point... It could be hidden up until a specific date before camp or so.

It just needs to be unique for each extension and of a reasonable length (10 digits?)

If this is doable we could just pull a CSV file with the extensions and verification numbers a week before the camp and have everything else prepared.

### PR DESCRIPTION
**Pull request by tykling**
DECT self-registration and phonebook functionality.

Branch is deployed to staging for testing: https://wwwstaging.bornhack.org/bornhack-2020/phonebook/ with the usual credentials for staging

Missing stuff:
- [x] API for POC team to download list of numbers and activation codes
- [ ] Does it make sense to somehow limit access to the oauth management views under /o/ ? @valberg ? 

This fixes #448 and part of #334

### CODE REVIEW

File: src/phonebook/templates/dectregistration_delete.html [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  <div class="panel-heading">
+    <span class="h3">Really delete DECT registration {{ dectregistration.number }}?</span>
+  </div>
+  <div class="panel-body">
+    <p class="lead">Are you sure you want to delete the registration of DECT number {{ dectregistration.number }}?
+    <ul>
+        <li>Letters: {{ dectregistration.letters|default:"N/A" }}</li>
+        <li>Description: {{ dectregistration.description|default:"N/A" }}</li>
+    </ul>
+    <form method="post" enctype="multipart/form-data">
```
Comment (valberg) [inline src/phonebook/templates/dectregistration_delete.html]: This doesn't strictly need the `enctype="multipart/form-data"` right?

File: src/profiles/admin.py [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 
 @admin.register(Profile)
-class OrderAdmin(admin.ModelAdmin):
```
Comment (valberg) [inline src/profiles/admin.py]: lol

File: src/utils/templatetags/menubutton.py [hunk 3]
```diff
@@ -12,3 +12,12 @@ def menubuttonclass(context, appname):
         return "btn-primary"
     else:
         return "btn-default"
+
+
+@register.simple_tag(takes_context=True)
+def menudropdownclass(context, appname):
```
Comment (valberg) [inline src/utils/templatetags/menubutton.py]: Nice

**Comment by valberg** [comment]
@tykling As discussed on IRC, I guess it's best just to proceed without limiting the oauth management views. It could be cool to give people the possibility to built things up against the website.

File: src/phonebook/templates/dectregistration_delete.html (hunk 1 above)
Comment (tykling) [reply src/phonebook/templates/dectregistration_delete.html]: right, copy/pasteerror :)

**Pull request merged**